examples/Kindle2Folder.rb

```diff
@@ -0,0 +1,29 @@
+require 'kindle_highlights'
+
+def sanitize_filename(filename) #Function to remove invalid symbols from filenames
+  fn = filename.split /(?<=.)\.(?=[^.])(?!.*\
```

Comment:
I think you're missing some characters here. Have you tried to run this in an irb console? What you want is something like:
```ruby
def sanitize_filename(filename)
  filename
    .split(/(?<=.)\.(?=[^.])(?!.*\.[^.])/m)
    .map! { |s| s.gsub /[^a-z0-9\-]+/i, '_' }
    .join('.')
end
```